WordPress/gutenberg
# Button: Add opt-in prop for next 40px default size

Part of #46741

## What?

Adds a `__next40pxDefaultSize` prop to opt into the new 40px default size for Button.

We're not quite ready to feature this prop in the documentation yet (like README and default code snippets), so I'll update those later when appropriate.

## Why?

We need a way for folks to gradually adopt the new sizing scheme in a controlled way.

## Testing Instructions

1. `npm run storybook:dev` and go to the Canvas view for the Button component.
2. Enable the Margin Checker tool in the toolbar for easier visualization.
3. For each of the Button stories, toggle the control for `__next40pxDefaultSize` and check that the size changes as expected.

https://user-images.githubusercontent.com/555336/235715283-39d96dbf-db8f-4f96-8081-5d3a77e7731f.mp4
